sunspec/core/util.py

```diff
@@ -95,6 +96,12 @@ def data_to_double(data):
         return d[0]
 
 def data_to_str(data):
+
+    # Change the data from bytes string to regular string for python 3
+    # compatibility
+    if sys.version_info > (3,):
+        data = str(data, 'latin-1')
```

Comment:
To your other points:

Latin-1 will be incorrect if, for example, my SunSpec device has a read-only string encoded in Shift-JIS.  At best I will get a decode exception, at worst I will silently be given garbage.  https://repl.it/JY0T/2  Further, not decoding in Python 2 and doing the decoding in Python 3 is inconsistent.

Since `str` in Python 2 is a byte string the compatibility libs are fairly compatible between their `bytes` and the Python 2 `str`s.  https://repl.it/JYYk/2  But, you get to write your code using the Python 3 terms `bytes` (for a sequence of bytes) and `str` (for a sequence of Unicode code points) and have it work the same way in both Python 3 and Python 2 (without `if` statements).  But again, this is only relevant if we are decoding which we should only do if either the spec is changed to define a single valid codec for strings or the user tells us what codec to use.  We should not guess because we will be wrong in various cases.  If we do decode/encode we should do so in both Python 2 and Python 3.

Yes, my example was moving between `unicode` and `str` (the byte sequence).  My point was that decoding and encoding are not Python 3 concepts.  They exist in Python 2 as well and they have different results with different codecs.  This library should do the same thing in both Python 2 and Python 3.  Either both decode/encode, or neither decodes/encodes.  Given that the Python 2 implementation did not along with us not knowing what codec would be correct, that's a double vote for not handling codings and just leaving it to the application.  Letting the `string` SunSpec type be just a byte sequence (with a little null termination below).